src/seed_tools/utils/version.test.ts

```diff
@@ -0,0 +1,223 @@
+          expect(() => new Version('01.1')).toThrowError();
+        });
+
+        it('should allow in subsequent components and ignore them', () => {
+          const v1 = new Version('1.01');
+          expect(v1.toString()).toEqual('1.1');
+          const v2 = new Version('1.1');
+          expect(v1).toEqual(v2);
+        });
+
+        it('should compare versions correctly when leading zeros are ignored', () => {
+          expect(new Version('1.02').gt(new Version('1.1'))).toBe(true);
+        });
+      });
+
+      describe('disallow', () => {
+        it('should not allow leading zeroes everywhere', () => {
+          const testCases = ['01.1', '1.01'];
+          testCases.forEach((testCase) => {
+            expect(
+              () => new Version(testCase, { disallowLeadingZeros: true }),
+            ).toThrowError();
+          });
+        });
+      });
+    });
+  });
+
+  describe('compare', () => {
+    const testCases = [
+      { lhs: '1.0', rhs: '1.0', expected: 0 },
+      { lhs: '1.0', rhs: '0.0', expected: 1 },
+      { lhs: '1.0', rhs: '2.0', expected: -1 },
+      { lhs: '1.0', rhs: '1.1', expected: -1 },
+      { lhs: '1.1', rhs: '1.0', expected: 1 },
+      { lhs: '1.0', rhs: '1.0.1', expected: -1 },
+      { lhs: '1.1', rhs: '1.0.1', expected: 1 },
+      { lhs: '1.1', rhs: '1.0.1', expected: 1 },
+      { lhs: '1.0.0', rhs: '1.0', expected: 0 },
+      { lhs: '1.0.3', rhs: '1.0.20', expected: -1 },
+      { lhs: '11.0.10', rhs: '15.007.20011', expected: -1 },
+      { lhs: '11.0.10', rhs: '15.5.28.130162', expected: -1 },
+      { lhs: '15.5.28.130162', rhs: '15.5.28.130162', expected: 0 },
+    ];
+
+    function expectCompare(lhs: Version, rhs: Version, expected: number) {
+      expect(lhs.compare(rhs)).toBe(expected);
+      expect(rhs.compare(lhs)).toBe(-expected + 0);
+
+      switch (expected) {
+        case -1:
+          expect(lhs.lt(rhs));
+          expect(lhs.lte(rhs));
+          expect(!lhs.eq(rhs));
+          expect(lhs.compare(rhs)).not.toBe(0);
+          expect(!lhs.gte(rhs));
+          expect(!lhs.gt(rhs));
+          break;
+        case 0:
+          expect(!lhs.lt(rhs));
+          expect(!lhs.lte(rhs));
+          expect(lhs.eq(rhs));
+          expect(lhs.compare(rhs)).toBe(0);
+          expect(!lhs.gte(rhs));
+          expect(!lhs.gt(rhs));
+          break;
+        case 1:
+          expect(!lhs.lt(rhs));
+          expect(!lhs.lte(rhs));
+          expect(!lhs.eq(rhs));
+          expect(lhs.compare(rhs)).not.toBe(0);
+          expect(lhs.gte(rhs));
+          expect(lhs.gt(rhs));
+          break;
+      }
+    }
+
+    testCases.forEach((testCase) => {
+      it(`should compare ${testCase.lhs} and ${testCase.rhs}`, () => {
+        const lhs = new Version(testCase.lhs);
+        const rhs = new Version(testCase.rhs);
+        expectCompare(lhs, rhs, testCase.expected);
+      });
+    });
+
+    describe('wildcard', () => {
+      const testCases = [
```

Comment:
nit: if we list test cases in ascending order we could just compare each pair, don't need to set specify `expected`

Reply:
kept this as is. Most of these tests are direct conversion from `base/version_unittest.cc`.